### ISSUE TITLE
indexpack: Fix a compatibility issue with the zip command-line tool. [phab:D30]

### ISSUE THREAD

**Issue by kythe** [google/kythe#229]
_Created by **fromberger** at 2015-02-06 19:24:19_: 

The zip command-line tool appends a trailing delimiter (usually "/") to
directory paths when creating an archive.  We weren't handling this right.
This change ensures we do, and updates the tests to cover it.

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2015-02-06 21:34:32_:

Why not

```
  needle := strings.TrimPrefix(path, z.prefix)
  dirNeedle := needle + string(filepath.Separator)
```

?

(Maybe someday we'll be ported to an operating system with a multi-byte  path separator.)

**Comment by kythe** [comment]
_**zarko** accepted this revision at 2015-02-06 21:34:32_

**Comment by kythe** [comment]
_**fromberger** wrote at 2015-02-06 22:06:39_:

  - Drop unwarranted premature optimization. :)

**Comment by kythe** [comment]
_**fromberger** closed via commit 95923f790ea49d1b3cfff075fe5b905a5905da1b at 2015-02-06 22:11:09_

**Comment by kythe** [comment]
_**fromberger** closed via commit 7d6bc028871f447fb035d159e9977448bfca4624 at 2015-02-06 22:11:09_

**Comment by kythe** [comment]
commit 6e93c36f94c0aa2592448bd27ac64e95476d6fcf
